controllers/certificate/certificate_controller_test.go

```diff
@@ -0,0 +1,233 @@
+package certificate
```

Comment:
To make sure we only test the public api, it would be good to use a test package. We don't have this for some legacy test files, but we should fore this for new tests.

```suggestion
package certificate_test
```